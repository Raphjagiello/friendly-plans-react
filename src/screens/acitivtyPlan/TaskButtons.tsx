import React from 'react';
import { StyleSheet, View } from 'react-native';

import { palette } from 'styles';
import { FlatButton, IconButton } from '../../components';

export class TaskButtons extends React.PureComponent {
  render() {
    return (
      <View style={styles.container}>
        <FlatButton
          title="TASUJÂ ZADANIA"
          icon={{
            name: 'shuffle',
            type: 'material-community-icons',
            color: palette.textBlackMuted,
            size: 22,
          }}
          buttonStyle={styles.buttonStyle}
          titleStyle={styles.buttonTitle}
        />
        <IconButton name="play" type="antdesign" size={36} color={palette.textBlackMuted} />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
  },
  buttonStyle: {
    marginRight: 20,
  },
  buttonTitle: {
    color: palette.textBlackMuted,
  },
});
