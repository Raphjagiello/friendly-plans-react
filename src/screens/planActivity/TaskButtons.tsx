import React from 'react';
import { StyleSheet, View } from 'react-native';

import { i18n } from 'locale';
import { palette } from 'styles';
import { FlatButton, IconButton } from '../../components';

export class TaskButtons extends React.PureComponent {
  render() {
    return (
      <View style={styles.container}>
        <FlatButton
          title={i18n.t('planActivity:shuffleTasks')}
          icon={{
            name: 'shuffle',
            type: 'material-community-icons',
            color: palette.textDisabled,
            size: 22,
          }}
          buttonStyle={styles.buttonStyle}
          titleStyle={styles.buttonTitle}
        />
        <IconButton name="play-circle" size={36} color={palette.textDisabled} />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  buttonStyle: {
    marginRight: 20,
  },
  buttonTitle: {
    color: palette.textDisabled,
    textTransform: 'uppercase',
  },
});
