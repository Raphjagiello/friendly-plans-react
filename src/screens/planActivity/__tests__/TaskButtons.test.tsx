import { shallow } from 'enzyme';
import React from 'react';
import { TaskButtons } from '../TaskButtons';

describe('TaskButtons', () => {
  it('renders correctly', () => {
    const taskButtons = shallow(<TaskButtons />);

    expect(taskButtons).toMatchSnapshot();
  });
});
