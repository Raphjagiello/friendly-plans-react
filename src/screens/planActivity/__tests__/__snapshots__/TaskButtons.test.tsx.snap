// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskButtons renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <FlatButton
    buttonStyle={
      Object {
        "marginRight": 20,
      }
    }
    icon={
      Object {
        "color": "#9B9B9B",
        "name": "shuffle",
        "size": 22,
        "type": "material-community-icons",
      }
    }
    title="Shuffle tasks"
    titleStyle={
      Object {
        "color": "#9B9B9B",
        "textTransform": "uppercase",
      }
    }
  />
  <IconButton
    color="#9B9B9B"
    name="play"
    size={36}
    type="antdesign"
  />
</View>
`;
